runtime/src/bank/check_transactions.rs

```diff
@@ -97,18 +99,24 @@ impl Bank {
             .zip(lock_results)
             .map(|(tx, lock_res)| match lock_res {
                 Ok(()) => {
+                    let feature_set: &FeatureSet = &self.feature_set;
+                    let transaction_fee = calculate_transaction_fee(
+                        tx.borrow(),
+                        self.fee_structure.lamports_per_signature,
+                        FeeFeatures::from(feature_set),
```

Comment:
could we move the definition of this outside the loop? features don't change for each transaction, so we can read the feature-set once and re-use the fee-specific bools.